opm/grid/polyhedralgrid/geometry.hh

```diff
@@ -41,6 +45,70 @@ namespace Dune
     typedef Dune::FieldVector< ctype, coorddimension > GlobalCoordinate;
     typedef Dune::FieldVector< ctype, mydimension >    LocalCoordinate;
 
+    typedef typename Grid::Traits::ExtraData  ExtraData;
+    typedef typename Grid::Traits::template Codim<codimension>::EntitySeed EntitySeed;
+
+    template< class ct >
+    struct PolyhedralMultiLinearGeometryTraits
+      : public Dune::MultiLinearGeometryTraits< ct >
+    {
+      struct Storage
+      {
+        struct Iterator
+          : public Dune::ForwardIteratorFacade< Iterator, GlobalCoordinate, GlobalCoordinate >
+        {
+          const Storage* data_;
+          int count_;
+          explicit Iterator( const Storage* ptr, int count ) : data_( ptr ), count_( count ) {}
+
+          GlobalCoordinate dereference() const { return data_->corner( count_ ); }
+          void increment() { ++count_; }
+
+          bool equals( const Iterator& other ) const { return count_ == other.count_; }
+        };
+
+        ExtraData  data_;
+        // host geometry object
+        EntitySeed seed_;
+
+        Storage( ExtraData data, EntitySeed seed )
+          : data_( data ), seed_( seed )
+        {}
+
+        Storage( ExtraData data )
+          : data_( data ), seed_()
+        {}
+
+        ExtraData data() const { return data_; }
```

Comment:
Should this rather return a const reference? I imagine ExtraData is not something you want to copy for every call to corner() or center()?

Reply:
ExtraData is of type Grid* so this is no problem (see grid.hh). EntitySeed is essentially the element index or similar so copying this is also fine.

Reply:
Ok, if you want this merged now that is fine with me.